### PR TITLE
chore: Update remaining dependencies - August 2022

### PR DESCRIPTION
**Pull request by HenrikSommer**
## Description

With this pull request, we update the remaining dependencies not updated in [1521](https://github.com/Energinet-DataHub/geh-charges/pull/1521) identified by dependabot.

3rd party packages updated:
- [X] AspNetCore.HealthChecks.AzureServiceBus -> 6.0.4
- [X] Azure.Messaging.ServiceBus -> 7.9.0
- [X] NodaTime.Testing -> 3.1.2

Energinet NuGet packages that are updated:
- [X] Energinet.DataHub.Core.FunctionApp.TestCommon -> 3.2.2
- [X] Energinet.DataHub.Core.Messaging -> 2.1.0
- [X] Energinet.DataHub.Core.Messaging.Protobuf -> 2.1.0
- [X] Energinet.DataHub.Core.Messaging.Protobuf.Integration.ServiceCollection -> 2.1.0
- [X] Energinet.DataHub.Core.Logging -> 2.2.0
- [X] Energinet.DataHub.Core.Schemas -> 2.1.0
- [X] Energinet.DataHub.Core.SchemaValidation -> 1.0.13 (not updated to 2.1.0 due to a conflict with Energinet.DataHub.Core.Messaging 2.1.0)
- [X] Energinet.DataHub.Core.TestCommon -> 3.2.2

Packages that will be updated in an upcoming pull request:
- Energinet.DataHub.MarketParticipant.Integration.Model

The project `GreenEnergyHub.Json` is removed and replaced by:
- [X] Energinet.DataHub.Core.JsonSerialization

Other remarks:
- `InstrumentationKey` is no longer explicitly provided as part of the update of `Microsoft.ApplicationInsights.WorkerService`
- `MessageMetaDataMiddleware.GetSessionId()` is changed to get `SessionId` from the BindingData key `SessionId` in contract to `MessageSession` due to changes in _Microsoft.Azure.Functions.Worker.Extensions.ServiceBus_ 5.4.0 

## References

* #1548

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure.Core/MessagingExtensions/Factories/ServiceBusMessageFactory.cs [hunk 1]
```diff
@@ -42,13 +42,24 @@ public ServiceBusMessage CreateInternalMessage(string data)
                     ApplicationProperties =
                     {
                         new KeyValuePair<string, object>("ReplyTo", _messageMetaDataContext.ReplyTo),
+                        new KeyValuePair<string, object>("OperationCorrelationId", _correlationContext.Id),
+                        // Todo: Integration test, set:
+                        // MessageTypeName = "
+                        // MessageVersionName
+                        // TimeStampName = "Op
+                        // CorrelationIdName =
+                        // EventIdentifierName
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure.Core/MessagingExtensions/Factories/ServiceBusMessageFactory.cs:51]:
```suggestion
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.FunctionHost/GreenEnergyHub.Charges.FunctionHost.csproj [hunk 2]
```diff
@@ -25,23 +25,23 @@ limitations under the License.
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="AspNetCore.HealthChecks.SqlServer" Version="6.0.2" />
-    <PackageReference Include="AspNetCore.HealthChecks.AzureServiceBus" Version="6.0.3" />
-    <PackageReference Include="Energinet.DataHub.Core.App.Common.Security" Version="4.0.2" />
-    <PackageReference Include="Energinet.DataHub.Core.App.FunctionApp" Version="3.0.1" />
-    <PackageReference Include="Energinet.DataHub.Core.Logging" Version="1.2.1" />
-    <PackageReference Include="Energinet.DataHub.Core.Messaging.Protobuf.Integration.ServiceCollection" Version="2.0.0" />
+    <PackageReference Include="AspNetCore.HealthChecks.AzureServiceBus" Version="6.0.4" />
+    <PackageReference Include="Energinet.DataHub.Core.App.Common.Security" Version="4.1.0" />
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.FunctionHost/GreenEnergyHub.Charges.FunctionHost.csproj:29]: Should `Energinet.DataHub.Core.App.Common.Security` have been version 5.0.0?

Comment (HenrikSommer) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.FunctionHost/GreenEnergyHub.Charges.FunctionHost.csproj:29]: 5.0.1, actually! It's a moving target!

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/EndpointTests/MeteringPointPersisterEndpointTests.cs [hunk 3]
```diff
@@ -137,19 +135,16 @@ public static string RandomString(int length)
...
-                parentId = $"00-{correlationId}-b7ad6b7169203331-01";
 
                 var byteArray = message.ToByteArray();
-                var serviceBusMessage = new ServiceBusMessage(byteArray)
-                {
-                    CorrelationId = correlationId,
-                };
+                var serviceBusMessage = new ServiceBusMessage(byteArray) { CorrelationId = correlationId };
                 serviceBusMessage.ApplicationProperties.Add("OperationTimestamp", date.ToUniversalTime());
                 serviceBusMessage.ApplicationProperties.Add("OperationCorrelationId", "1bf1b76337f14b78badc248a3289d021");
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/EndpointTests/MeteringPointPersisterEndpointTests.cs:142]: nit:  Not part of your change, but should the value have been CorrelationId here?
```suggestion
                serviceBusMessage.ApplicationProperties.Add("OperationCorrelationId", CorrelationId);
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.WebApi/GreenEnergyHub.Charges.WebApi.csproj [hunk 4]
```diff
@@ -22,9 +22,9 @@ limitations under the License.
 
   <ItemGroup>
     <PackageReference Include="AspNetCore.HealthChecks.SqlServer" Version="6.0.2" />
-    <PackageReference Include="Energinet.DataHub.Core.App.Common" Version="4.0.2" />
-    <PackageReference Include="Energinet.DataHub.Core.App.Common.Security" Version="4.0.2" />
-    <PackageReference Include="Energinet.DataHub.Core.App.WebApp" Version="2.3.3" />
+    <PackageReference Include="Energinet.DataHub.Core.App.Common" Version="5.0.0" />
+    <PackageReference Include="Energinet.DataHub.Core.App.Common.Security" Version="5.0.1" />
+    <PackageReference Include="Energinet.DataHub.Core.App.WebApp" Version="4.1.0" />
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.WebApi/GreenEnergyHub.Charges.WebApi.csproj:27]: ?
```suggestion
    <PackageReference Include="Energinet.DataHub.Core.App.WebApp" Version="5.0.1" />
```

**Review by prtandrup — APPROVED** [review]
LGTM with a couple of comments.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/EndpointTests/MeteringPointPersisterEndpointTests.cs (hunk 3 above)
Comment (HenrikSommer) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/EndpointTests/MeteringPointPersisterEndpointTests.cs:142]: Well spotted! I will fix that in pull request #1567

**Pull request merged**